### ISSUE TITLE
remove the --no-config flag for ripgrep

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#31530]
TL;DR As a developer I want to use the ripgrep config file to tailor searching large repositories so I can get the results that I need
-----
vscode uses ripgrep to perform search operations. When launching ripgrep, vscode passes a flag --no-config to not interpret any settings a user may have in a ripgrep-config file. Any folders to exclude from the search are also passed as flag to ripgrep.

These are the places where the no-config flag is set:
https://github-com/search?q=repo%3Amicrosoft%2Fvscode%20--no-config&type=code

Where there is a large number of folders to exclude from the search (e.g. through .gitnore settings or search.exclude settings in the vscode config), launching ripgrep may give a ENAMETOOLONG error (since the total command is too long for the shell to handle).

See issue -869 One reason to have a long list of folders to exclude, may come from vscode inability to efficiently exclude certain folders from search (and the explorer for that matter). Could we use the ripgrep config file, we could more efficiently exclude folders from the search.

Note 1: it would be *even* better if vscode itself would use the config file to exclude folders (and maintain it), but I can see that that feature request would also need the same syntax support to exclude folders from the file explorer and that may be a bit too much to ask for.
Note 2: I'd be happy to create a PR for removing the --no-config flag.
